system-addon/common/ShortURL.jsm

```diff
@@ -1,7 +1,10 @@
 Components.utils.importGlobalProperties(["URL"]);
 
+const Cc = Components.classes;
+const Ci = Components.interfaces;
```

Comment:
This seems a bit odd to put `Components` usage under `common/`. Other usage seems to have various guards or is importing an activity-stream jsm so webpack converts it. Did you look at putting the IDNService usage under `lib/` ?